File: sdks/jvm/src/main/java/ai/condux/SendResult.java
package ai.condux;

/**
 * Outcome of a delivery attempt sequence. Never thrown — inspect {@link #ok()}. {@code status} is null
 * when every attempt failed at the network level (no HTTP response); {@code error} carries the last
 * network failure message in that case.
 */
public record SendResult(boolean ok, int attempts, Integer status, String error) {
}

File: sdks/dotnet/Condux.Sdk/SendResult.cs
namespace Condux.Sdk;

/// <summary>The outcome of a delivery attempt sequence. Reporting never throws — inspect
/// <see cref="Ok"/>. <see cref="Status"/> is the last HTTP status seen (null if every attempt was a
/// network error); <see cref="Error"/> is the last transport error message.</summary>
public sealed record SendResult(bool Ok, int Attempts, int? Status = null, string? Error = null);
